src/get_package_version.py

```diff
@@ -1,19 +1,83 @@
-from typing import Final
-from pathlib import Path
 import logging
 import subprocess
+import functools
+import re
+from typing import Final
+from pathlib import Path
+from packaging import version, specifiers
 
-from src.module import (
-    get_module_name,
-    get_module_version,
-)
+from src.module import get_module_name
 
 logger = logging.getLogger(__name__)
-
+logger.setLevel(logging.DEBUG)
 # version attribute as listed in apt-cache show
 VERSION_ATTRIBUTE: Final = "Version"
 VERSION_DOT_TXT: Final = Path("version.txt")
 
+class FileSystem(object):
+    def path_exists(self, path: Path) -> bool:
+        return path.exists()
+
+    def get_directories(self, path: Path) -> list:
+        return [version.name for version in path.iterdir() if version.is_dir()]
+
+    def get_file_contents(self, path: Path) -> str:
+        with open(path, "r", encoding='utf-8') as file:
+            return file.read()
+
+
+def satisfies_specification(pkg_version: str, spec: str) -> bool:
+    logger.debug("Checking if package version %s satisfies specification %s", pkg_version, spec)
+    v = version.parse(pkg_version)
+    
+    return v in specifiers.SpecifierSet(spec)
+
+def parse_debian_version(version_string: str):
+    # https://www.debian.org/doc/debian-policy/ch-controlfields.html#s-f-version
+    match = re.match(r"(?:(\d+):)?((?:[^-]|-(?!\d))*)(?:-([\d].*))?", version_string)
+    if match is None:
+        raise ValueError(f"Invalid Debian version string: {version_string}")
+    epoch, debian_version, _ = match.groups()
+    try:
+        epoch = int(epoch)
+    except TypeError:
+        # epoch is not a correct integer
+        epoch = None
+
+    # Process debian_version to make it compatible with packaging.version.parse
+    debian_version = debian_version.split('~')[0]
+    debian_version = debian_version.split('+')[0]
+    debian_version = debian_version.split('-')[0]
+    debian_version = re.split('[^0-9.]', debian_version)[0].rstrip('.')
+
+    return epoch, debian_version
+
+def compare_debian_versions(version1: str, version2: str) -> int:
```

Comment:
maybe return an enum here ?
Like LogicalOperator enum that has LARGER_THAN, ... etc